R/report_parameters.R

```diff
@@ -175,36 +175,20 @@ print.report_parameters <- function(x, ...) {
 
     if ("ESS" %in% names(diagnostic)) {
       stability <- effectsize::interpret_ess(diagnostic$ESS, ...)
-      text <- ifelse(stability == "sufficient" & convergence == "converged",
-        paste0(
-          text,
-          " and the indices are reliable (ESS = ",
-          insight::format_value(diagnostic$ESS, digits = 0),
-          ")"
-        ),
-        ifelse(stability == "sufficient" & convergence != "converged",
-          paste0(
-            text,
-            " even though the indices appear as reliable (ESS = ",
-            insight::format_value(diagnostic$ESS, digits = 0),
-            ")"
-          ),
-          ifelse(stability != "sufficient" & convergence == "converged",
-            paste0(
-              text,
-              " but the indices are unreliable (ESS = ",
-              insight::format_value(diagnostic$ESS, digits = 0),
-              ")"
-            ),
-            paste0(
-              text,
-              " and the indices are unreliable (ESS = ",
-              insight::format_value(diagnostic$ESS, digits = 0),
-              ")"
-            )
-          )
+      text <- lapply(seq_len(length(stability)), function(x) {
+        y <- switch(EXPR = paste(stability[x] == "sufficient", convergence[x] == "converged"),
+          "TRUE TRUE" = paste0(text, " and the indices are reliable (ESS = ",
+                               insight::format_value(diagnostic$ESS, digits = 0), ")"),
+          "TRUE FALSE" = paste0(text, " even though the indices appear as reliable (ESS = ",
+                                insight::format_value(diagnostic$ESS, digits = 0), ")"),
+          "FALSE TRUE" = paste0(text, " but the indices are unreliable (ESS = ",
+                                insight::format_value(diagnostic$ESS, digits = 0), ")"),
+          "FALSE FALSE" = paste0(text, " and the indices are unreliable (ESS = ",
+                                 insight::format_value(diagnostic$ESS, digits = 0), ")")
         )
-      )
+        y[[x]]
+      })
+      text <- unlist(text)
```

Comment:
Not a fan of such type of conversion (boolean -> string/character), but let's keep this for now and revisit once I have something better to offer.

Reply:
In this case, I was attempting to get rid of the nested `ifelse`s and wanted to apply switch as per easystats/easystats#335. The challenge was that I had an error saying that switch cannot work with an object of length > 1. And it seemed to be the same for other strategies (lists, look-up data.frame, etc.). That was the best I could come up with 😬